File: fact.c
/*
    @@@ get_version: gcc --version 2>/dev/null| grep -E 'gcc|version'
    @@@ before: gcc -o fact_c fact.c
    @@@ instead: ./fact_c
    @@@ after: rm fact_c
    @@@ is_fast: 1
*/

#include <stdio.h>

#define times 100000000
#define fact_16 20922789888000

long fact(int n) {
    if (n == 1)
        return 1;
    return(n * fact(n - 1));
}

int main() {
    int ok = 1;
    long i;
    for(i = 0; i < times; ++i) {
        ok = ok && fact(16) == fact_16;
    }
    printf("C finish %i - %s\n", times, ok ? "ok" : "fail");
}

File: fact.kt
/*
    @@@ get_version: docker run --rm zenika/kotlin kotlinc -version 2>&1 | awk '$2 == "kotlinc-jvm" {print $3}'
    @@@ before: docker run --rm -v "$PWD":/app -w /app zenika/kotlin kotlinc fact.kt
    @@@ instead: docker run --rm -v "$PWD":/app -w /app zenika/kotlin kotlin FactKt
    @@@ after: rm -rf FactKt.class ./META-INF
    @@@ is_fast: 1
*/

private val times = 200000000
private val fact16 = 20922789888000L

fun fact(n: Long): Long {
    if (n == 1L)
        return 1
    return n * fact(n - 1)
}

fun main() {
    var ok: Boolean = true

    for (i: Int in 1..times) {
        ok = ok && fact(16) == fact16
    }

    val result = if (ok) "ok" else "fail"
    println("Kotlin finish $times - $result")
}
